code/modules/mob/living/carbon/superior_animal/superior_animal.dm

```diff
@@ -372,12 +386,19 @@
 	if(!ranged)
 		prepareAttackOnTarget()
 	else if(ranged)
+		if(!check_if_alive())
+			return
 		if(get_dist(src, targetted_mob) <= 6)
-			OpenFire(targetted_mob)
+			addtimer(CALLBACK(src, .proc/OpenFire, targetted_mob), delay_for_range)
 		else
 			set_glide_size(DELAY2GLIDESIZE(move_to_delay))
 			walk_to(src, targetted_mob, 4, move_to_delay)
-			OpenFire(targetted_mob)
+			addtimer(CALLBACK(src, .proc/OpenFire, targetted_mob), delay_for_range)
+
+/mob/living/carbon/superior_animal/proc/check_if_alive() //A simple yes no if were alive
```

Comment:
unneeded proc, use is_dead()

Reply:
keeping it even more simple then is_dead